### PR TITLE
chore: check if AIGateway CRD exists in the cluster before creating a count provider

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

- Prevent users being confused by the error message being logged when AIGateway count provider fails to get created

  Related issue: #297 

- Additionally this creates a CRD guard so that when user requests AIGateway controller but the CRD is not in place then controller exits with an error.

- This also fixes the aigateway reports in telemetry. For some reason (probably due to the incorrect pluralized version `aigatewies` used in AIGateway GVR).

  The reports now include the AIGateway entry:
  
  ```
  report: <14>signal=gateway-operator-start;hn=P-Maek-MBP;uptime=0;v=NOT_SET;k8s_aigateways_count=1;k8s_controlplanes_count=0;k8s_controlplanes_requested_replicas_count=0;k8s_dataplanes_count=0;k8s_dataplanes_requested_replicas_count=0;k8s_gatewayclasses_count=1;k8s_gateways_count=0;k8s_httproutes_count=0;k8s_nodes_count=1;k8s_pods_count=14;k8s_referencegrants_count=0;k8s_services_count=4;k8s_standalone_controlplanes_count=0;k8s_standalone_dataplanes_count=0;k8s_arch=linux/arm64;k8s_provider=kind;k8sv=v1.30.0;k8sv_semver=v1.30.0;
  ```

### Special notes for your reviewer:

For whatever reason the tests do not allow the full suite to pass as it used to. It's most likely relate to the pluralising issue in k8s: https://github.com/kubernetes/client-go/issues/1082

### CODE REVIEW

File: internal/telemetry/telemetry_test.go [hunk 1]
```diff
@@ -411,41 +437,65 @@ func TestTelemetryUpdates(t *testing.T) {
...
 	}
 	for _, tc := range testcases {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
 			scheme := prepareScheme(t)
+			// We need the custom list kinds to prevent:
+			// panic: coding error: you must register resource to list kind for every resource you're going
+			// to LIST when creating the client.
+			// See NewSimpleDynamicClientWithCustomListKinds or register the list into the scheme:
+			// gateway-operator.konghq.com/v1alpha1, Resource=aigateways out of map
```
Comment (rainest) [inline internal/telemetry/telemetry_test.go]:
```suggestion
			// See NewSimpleDynamicClientWithCustomListKinds:
			// https://pkg.go.dev/k8s.io/client-go/dynamic/fake#NewSimpleDynamicClientWithCustomListKinds
			// or register the list into the scheme:
```

File: internal/telemetry/telemetry_test.go [hunk 2]
```diff
@@ -330,7 +344,17 @@ func TestCreateManager(t *testing.T) {
 			require.True(t, ok)
 			d.FakedServerVersion = versionInfo()
 
-			dyn := testdynclient.NewSimpleDynamicClient(scheme, tc.objects...)
+			// We need the custom list kinds to prevent:
+			// panic: coding error: you must register resource to list kind for every resource you're going
+			// to LIST when creating the client.
+			// See NewSimpleDynamicClientWithCustomListKinds or register the list into the scheme:
+			// gateway-operator.konghq.com/v1alpha1, Resource=aigateways out of map
```
Comment (rainest) [inline internal/telemetry/telemetry_test.go]:
```suggestion
			// See NewSimpleDynamicClientWithCustomListKinds:
			// https://pkg.go.dev/k8s.io/client-go/dynamic/fake#NewSimpleDynamicClientWithCustomListKinds
			// or register the list into the scheme:
```

**Review by rainest — CHANGES_REQUESTED** [review]

**Comment by randmonkey** [comment]
Does this PR need a changelog entry?

File: internal/telemetry/telemetry_test.go [hunk 3]
```diff
@@ -411,41 +438,66 @@ func TestTelemetryUpdates(t *testing.T) {
...
 	for _, tc := range testcases {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
 			scheme := prepareScheme(t)
+			// We need the custom list kinds to prevent:
+			// panic: coding error: you must register resource to list kind for every resource you're going
+			// to LIST when creating the client.
+			// See NewSimpleDynamicClientWithCustomListKinds:
+			// https://pkg.go.dev/k8s.io/client-go/dynamic/fake#NewSimpleDynamicClientWithCustomListKinds
+			// or register the list into the scheme:
```
Comment (pmalek) [inline internal/telemetry/telemetry_test.go:492]: The list is also registered into the scheme https://github.com/Kong/gateway-operator/blob/3876430e09e61edce58bd8464989e33236bd1872/api/v1alpha1/aigateway_types.go#L254-L255. So it's not that.

The problems seems to be again, with pluralizing "gateway". When you remove this mapping (from line 496) you get all the mapping and you can see that we have the following in there:

```
...
Resource=aigatewaies:AIGatewayList gateway-operator.konghq.com/v1alpha1
...
```

**Comment by pmalek** [comment]
> Does this PR need a changelog entry?

@randmonkey No need for changelog, this is not user facing.

**Review by rainest — APPROVED** [review]

**Pull request merged**